legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/classpath/ClasspathUsingMavenFactory.java

```diff
@@ -446,28 +429,45 @@ private URLClassLoader createClassloader(File maven, File pom, File settingXml,
             LOGGER.error("Unable to construct extensions classpath", e);
             return null;
         }
+        finally
+        {
+            try
+            {
+                if (tempPom != null)
+                {
+                    Files.delete(tempPom);
+                }
+            }
+            catch (IOException e)
+            {
+                LOGGER.error("Failed to delete temp pom", e);
+            }
```

Comment:
You might try setting deleteOnExit if for some reason you can't delete the file.
```suggestion
            if (tempPom != null)
            {
                try
                {
                    Files.delete(tempPom);
                }
                catch (Exception e)
                {
                    LOGGER.error("Failed to delete temp pom: {}", tempPom, e);
                    try
                    {
                        tempPom.toFile().deleteOnExit();
                    }
                    catch (Exception ee)
                    {
                        LOGGER.warn("Failed to set deleteOnExit for temp pom: {}", tempPom, ee);
                    }
                }
            }
```

Reply:
I added this on the latest commit.